landoapi/models/landing.py

```diff
@@ -0,0 +1,115 @@
+from landoapi.phabricator_client import PhabricatorClient
+from landoapi.transplant_client import TransplantClient
+
+TRANSPLANT_JOB_STARTED = 'started'
+TRANSPLANT_JOB_FINISHED = 'finished'
+
+
+def _get_revision(revision_id, api_key=None):
+    """ Gets revision from Phabricator.
+
+    Returns None or revision.
+    """
+    phab = PhabricatorClient(api_key)
+    revision = phab.get_revision(id=revision_id)
+    if not revision:
+        return None
+
+    raw_repo = phab.get_repo(revision['repositoryPHID'])
+    return {
+        'id': int(revision['id']),
+        'phid': revision['phid'],
+        'repo_url': raw_repo['uri'],
+        'title': revision['title'],
+        'url': revision['uri'],
+        'date_created': int(revision['dateCreated']),
+        'date_modified': int(revision['dateModified']),
+        'status': int(revision['status']),
+        'status_name': revision['statusName'],
+        'summary': revision['summary'],
+        'test_plan': revision['testPlan'],
+    }
+
+
+class Landing(db.Model):
+    __tablename__ = "landings"
+
+    id = db.Column(db.Integer, primary_key=True)
+    request_id = db.Column(db.Integer, unique=True)
+    revision_id = db.Column(db.String(30))
+    status = db.Column(db.Integer)
+
+    def __init__(
+        self, request_id=None, revision_id=None, status=TRANSPLANT_JOB_STARTED
+    ):
+        self.request_id = request_id
+        self.revision_id = revision_id
+        self.status = status
+
+    @classmethod
+    def create(cls, revision_id, phabricator_api_key=None, save=True):
+        """ Land revision and create a Transplant item in storage. """
+        revision = _get_revision(revision_id, phabricator_api_key)
+        if not revision:
+            raise RevisionNotFoundException()
+
+        trans = TransplantClient()
+        request_id = trans.land(
+            'ldap_username@example.com', revision['repo_url']
+        )
+        if not request_id:
+            raise LandingNotCreatedException
+
+        t = cls(
+            revision_id=revision_id,
+            request_id=request_id,
+            status=TRANSPLANT_JOB_STARTED
+        )
+        if save:
+            t.save(create=True)
+
+        return t
+
+    @classmethod
+    def get(cls, landing_id):
+        """ Get Landing object from storage. """
+        t = cls.query.get(landing_id)
+        if not t:
+            raise LandingNotFoundException()
+        return t
+
+    def save(self, create=False):
+        """ Save objects in storage. """
+        if create:
+            db.session.add(self)
+
+        db.session.commit()
+        return self
+
+    def __repr__(self):
+        return '<Landing: %s>' % self.id
+
+    def serialize(self):
+        """ Serialize """
+        return {
+            'id': self.id,
+            'revision_id': self.revision_id,
+            'request_id': self.request_id,
+            'status': self.status
+        }
+
+
+class LandingNotCreatedException(Exception):
+    pass
```

Comment:
Needs a docstring to explain that this exception represents downstream service errors.